instrumentation/opentelemetry-instrumentation-boto3sqs/src/opentelemetry/instrumentation/boto3sqs/__init__.py

```diff
@@ -0,0 +1,430 @@
+            )
+        if message_id:
+            span.set_attribute(SpanAttributes.MESSAGING_MESSAGE_ID, message_id)
+
+    @staticmethod
+    def _safe_end_processing_span(receipt_handle: str) -> None:
+        if started_span := Boto3SQSInstrumentor.received_messages_spans.pop(
+            receipt_handle, None
+        ):
+            if (
+                Boto3SQSInstrumentor.current_span_related_to_token
+                == started_span
+            ):
+                context.detach(Boto3SQSInstrumentor.current_context_token)
+                Boto3SQSInstrumentor.current_context_token = None
+            started_span.end()
+
+    @staticmethod
+    def _extract_queue_name_from_url(queue_url: str) -> str:
+        # A Queue name cannot have the `/` char, therefore we can return the part after the last /
+        return queue_url.split("/")[-1]
+
+    def _create_processing_span(
+        self, queue_name: str, receipt_handle: str, message: Dict[str, Any]
+    ) -> None:
+        message_attributes = message.get("MessageAttributes", {})
+        links = []
+        if ctx := propagate.extract(
+            message_attributes, getter=boto3sqs_getter
+        ):
+            for item in ctx.values():
+                if hasattr(item, "get_span_context"):
+                    links.append(Link(context=item.get_span_context()))
+        span = self._tracer.start_span(
+            name=f"{queue_name} process",
+            links=links,
+            kind=SpanKind.CONSUMER,
+        )
+        with trace.use_span(span):
+            message_id = message.get("MessageId", None)
+            Boto3SQSInstrumentor.received_messages_spans[receipt_handle] = span
+            Boto3SQSInstrumentor._enrich_span(
+                span,
+                queue_name,
+                message_id=message_id,
+                operation=MessagingOperationValues.PROCESS,
+            )
+
+    def _wrap_send_message(self) -> None:
+        def send_wrapper(wrapped, instance, args, kwargs):
+            if context.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
+                return wrapped(*args, **kwargs)
+            queue_url = kwargs.get("QueueUrl")
+            # The method expect QueueUrl and Entries params, so if they are None, we call wrapped to receive the
+            # original exception
+            queue_name = Boto3SQSInstrumentor._extract_queue_name_from_url(
+                queue_url
+            )
+            span = self._tracer.start_span(
+                name=f"{queue_name} send",
+                kind=SpanKind.PRODUCER,
+            )
+            Boto3SQSInstrumentor._enrich_span(span, queue_name)
+            with trace.use_span(span, end_on_exit=True):
+                attributes = kwargs.pop("MessageAttributes", {})
+                propagate.inject(attributes, setter=boto3sqs_setter)
+                retval = wrapped(*args, MessageAttributes=attributes, **kwargs)
+                if message_id := retval.get("MessageId"):
```

Comment:
you can pass message_id in enrich_span() so that you dont have to repeat the line 242 and 243 again as its already present in enrich_span()

Reply:
I think it's better to add it later, because if the call of `retval = wrapped(*args, MessageAttributes=attributes, **kwargs) ` will raise (by a  wrong queue for example), the span will still have the available information - helping the user to trace and debug. @sanketmehta28 WDYT?